include/seqan3/alignment/pairwise/detail/policy_optimum_tracker.hpp

```diff
@@ -0,0 +1,173 @@
+// -----------------------------------------------------------------------------------------------------
+// Copyright (c) 2006-2020, Knut Reinert & Freie Universität Berlin
+// Copyright (c) 2016-2020, Knut Reinert & MPI für molekulare Genetik
+// This file may be used, modified and/or redistributed under the terms of the 3-clause BSD-License
+// shipped with this file and also available at: https://github.com/seqan/seqan3/blob/master/LICENSE.md
+// -----------------------------------------------------------------------------------------------------
+
+/*!\file
+ * \brief Provides seqan3::detail::policy_optimum_tracker.
+ * \author Rene Rahn <rene.rahn AT fu-berlin.de>
+ */
+
+#pragma once
+
+#include <limits>
+
+#include <seqan3/alignment/pairwise/detail/type_traits.hpp>
+#include <seqan3/core/algorithm/configuration.hpp>
+#include <seqan3/core/type_traits/template_inspection.hpp>
+
+namespace seqan3::detail
+{
+
+/*!\brief Implements the tracker to store the global optimum for a particular alignment computation.
+ * \ingroup pairwise_alignment
+ *
+ * \tparam alignment_configuration_t The type of the alignment configuration; must be a type specialisation of
+ *                                   seqan3::configuration.
+ *
+ * \details
+ *
+ * Implements the interface to track the alignment optimum. It updates the currently stored optimum if the
+ * optimal score of the given cell compares greater or equal to the sotred optimum.
+ * Special methods are offered to track any cell (for example when computing the local alignment), the last cell of a
+ * column or a row (for example when using free-end gaps), or the final cell of the entire matrix (for example in the
+ * standard global alignment).
+ * Depending on the configuration they might or might not evaluate the stored optimum score of that cell to compare
+ * it with the currently stored maximum.
```

Comment:
```suggestion
 * it with the currently stored optimum.
```
for consistency with text above